client/network/tests/sync_blocks_from_tip_without_collator.rs

```diff
@@ -27,12 +28,24 @@ async fn sync_blocks_from_tip_without_being_connected_to_a_collator() {
 	let para_id = ParaId::from(100);
 	let tokio_handle = tokio::runtime::Handle::current();
 
+	let ws_port = portpicker::pick_unused_port().expect("No free ports");
```

Comment:
Hmm.... while the chance is probably low, this could still return a port which isn't really free.

If we'd try to listen on port `0` the OS will atomically pick a free port by itself, so could we perhaps set the `ws_port` to `0` when calling `run_relay_chain_validator_node` and then just read which port was picked?

Reply:
As I see it the portpicker has logic to let the OS pick the port. Or do you mean the port could become occupied after picking the port here but before the full-node is started?

Reply:
It can let the OS pick the port, but it immediately releases it, so it's inherently racy and something which runs in parallel (another thread, another program on the same machine, etc.) could use that port before we manage to start the full node.

Reply:
As discussed in the meeting, I do not see an easy way to get the port here right now. I will observe if we see failures due to this, but should be good for now.